store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/key/core/AbstractCoreKeyAccumuloElementConverter.java

```diff
@@ -117,50 +118,63 @@ public Value getValueFromElement(final Element element) {
     @Override
     public Properties getPropertiesFromValue(final String group, final Value value) {
         final Properties properties = new Properties();
-        if (value == null || value.getSize() == 0) {
-            return properties;
-        }
-        final byte[] bytes = value.get();
-        int lastDelimiter = 0;
-        final int arrayLength = bytes.length;
-        long currentPropLength;
-        final SchemaElementDefinition elementDefinition = getSchemaElementDefinition(group);
-        final Iterator<String> propertyNames = elementDefinition.getProperties().iterator();
-        while (propertyNames.hasNext() && lastDelimiter < arrayLength) {
-            final String propertyName = propertyNames.next();
-            if (isStoredInValue(propertyName, elementDefinition)) {
-                final TypeDefinition typeDefinition = elementDefinition.getPropertyTypeDef(propertyName);
-                final ToBytesSerialiser serialiser = (typeDefinition != null) ? (ToBytesSerialiser) typeDefinition.getSerialiser() : null;
-                if (null != serialiser) {
-                    final int numBytesForLength = CompactRawSerialisationUtils.decodeVIntSize(bytes[lastDelimiter]);
-                    final byte[] length = new byte[numBytesForLength];
-                    System.arraycopy(bytes, lastDelimiter, length, 0, numBytesForLength);
-                    try {
-                        currentPropLength = CompactRawSerialisationUtils.readLong(length);
-                    } catch (final SerialisationException e) {
-                        throw new AccumuloElementConversionException("Exception reading length of property", e);
-                    }
-                    lastDelimiter += numBytesForLength;
-                    if (currentPropLength > 0) {
-                        try {
-                            properties.put(propertyName, serialiser.deserialise(Arrays.copyOfRange(bytes, lastDelimiter, lastDelimiter += currentPropLength)));
-                        } catch (final SerialisationException e) {
-                            throw new AccumuloElementConversionException("Failed to deserialise property " + propertyName, e);
-                        }
-                    } else {
-                        try {
-                            properties.put(propertyName, serialiser.deserialiseEmpty());
-                        } catch (final SerialisationException e) {
-                            throw new AccumuloElementConversionException("Failed to deserialise property " + propertyName, e);
+        if (isNotEmpty(value)) {
+            final byte[] bytes = value.get();
+            int carriage = 0;
```

Comment:
We use the name 'lastDelimiter' in a few places and I'm not sure the word 'carriage' is actually any better, so please could we keep it as 'lastDelimiter'.